File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/rabbitmq/docs/properties/ProductListener.kt
package io.micronaut.rabbitmq.docs.properties

// tag::imports[]
import io.micronaut.rabbitmq.annotation.Queue
import io.micronaut.rabbitmq.annotation.RabbitListener
import io.micronaut.rabbitmq.annotation.RabbitProperty
import io.micronaut.context.annotation.Requires
import java.util.*
// end::imports[]

@Requires(property = "spec.name", value = "PropertiesSpec")
// tag::clazz[]
@RabbitListener
class ProductListener {

    val messageProperties: MutableList<String> = Collections.synchronizedList(ArrayList())

    @Queue("product")
    @RabbitProperty(name = "x-priority", value = "10", type = Integer::class) // <1>
    fun receive(data: ByteArray,
                @RabbitProperty("userId") user: String, // <2>
                @RabbitProperty contentType: String?, // <3>
                appId: String) { // <4>
        messageProperties.add("$user|$contentType|$appId")
    }
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/properties/ProductListener.py
from typing import Annotated

from micronaut.context.annotation import Requires
# tag::imports[]
from java.lang import Integer
from micronaut.rabbitmq.annotation import Queue, RabbitListener, RabbitProperty
# end::imports[]


@Requires(property="spec.name", value="PropertiesSpec")
# tag::clazz[]
@RabbitListener
class ProductListener:

    def __init__(self) -> None:
        self.message_properties: list[str] = []

    @Queue("product")
    @RabbitProperty(name="x-priority", value="10", type=Integer)  # <1>
    def receive(self, data: bytes,
                user: Annotated[str, RabbitProperty("userId")],  # <2>
                content_type: Annotated[str | None, RabbitProperty("contentType")],  # <3>
                appId: str) -> None:  # <4>
        self.message_properties.append(f"{user}|{content_type}|{appId}")
# end::clazz[]
